datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1438,9 +1436,31 @@ mod tests {
         let input_types = vec![DataType::Int32, DataType::Int32];
         let return_type = Arc::new(DataType::Int32);
 
-        let fun = |args: &[ArrayRef]| {
-            let arg0 = as_int32_array(&args[0])?;
-            let arg1 = as_int32_array(&args[1])?;
+        let fun = Arc::new(|args: &[ColumnarValue]| {
+            let len = args
```

Comment:
Maybe as a follow on PR we can make a function that does this length inference and conversion to array (mostly so it can be documented) to make the ideas easier to find 🤔

Reply:
Yea, have the same thought too. It will be still useful for `ScalarUDFImpl` as `invoke` takes `args: &[ColumnarValue]` and users might need length inference too.